src/nonlinearshrinkage.jl

```diff
@@ -51,39 +51,39 @@ http://www.econ.uzh.ch/static/wp/econwp264.pdf
 """
 function analytical_nonlinear_shrinkage(S::AbstractMatrix{<:Real},
                                         n::Int, p::Int, est_mean::Bool;
-                                        decomp::Union{Nothing,Eigen}=nothing)
+                                        decomp::Eigen=eigen(S))
 
+    η = ifelse(p < n, n, n - Int(est_mean)) # effective sample size
     # sample eigenvalues sorted in ascending order and eigenvectors
-    F    = isa(decomp, Nothing) ? eigen(S) : decomp # O(p^3)
+    F    = decomp # eigen is O(p^3)
     perm = sortperm(F.values)
-    λ    = F.values[perm]
+    sample_perm = @view perm[max(1, (p - η) + 1):p]
+    λ    = @view F.values[sample_perm]
     U    = F.vectors[:, perm]
```

Comment:
this one also could become a view, but i suspect it is not worth it, as it will break the striding of the array, which will make BLAS sad.

Reply:
I guess BLAS will also be sad about lambda being a view. Do these views actually make this function faster?

Reply:
λ is never used in any BLAS calls AFAICT.

I haven't benchmarked it. In general they do.